### DISCUSSION TITLE
Controller/service generation

### DISCUSSION THREAD

**Discussion by ashryanbeats** [oas-tools/oas-tools.github.io#15] [Q&A] [answered]
Hi oas-tools community! I'm really loving this tool and I'm so happy it exists as a tool for API design-first approaches to building APIs for Node.js Express.

Beyond the initial bootstrapping of a server, I'm curious if it's possible to continuously use oas-tools for route/controller/service _generation_ in a project.

Currently, if I go into my API spec and add a new path to the API design, next time I start up my server, I'll get help helpful error messages that note if I'm missing, say, a controller, service, or service method.

I haven't really thought this through yet, so I wanted to ask: would it be useful if oas-tools could offer to create missing files and service methods?

**Comment by alesancor1** [comment, marked as answer]
I'm not sure if the CLI tool generates missing files, but it could be interesting if it did, since you could set the ´start´ script in `package.json` generate the missing files through CLI before actually starting the server.

There are ways to implement it so I'll take a look, I think it is an interesting feature for future versions :)
